File: build-type/01_singleton/go/main.go
package main

import (
	"fmt"
)

func main() {

	// 饿汉式
	singleton := GetInstance()
	fmt.Printf("Eager Singleton: %p\n", singleton)

	// 懒汉式
	// 单例模式中获取到的对象始终是同一个对象
	// 即有相同的内存地址
	lazySingleton1 := GetLazyInstance()
	lazySingleton2 := GetLazyInstance()
	fmt.Printf("Lazy Singleton1: %p\n", lazySingleton1)
	fmt.Printf("Lazy Singleton2: %p\n", lazySingleton2)
}

File: build-type/01_singleton/java/Main.java
package main;

public class Main {

    public static void main(String[] args) {

        var singleton = SingletonEager.getInstance();
        System.out.println(singleton);

        var lazySingleton1 = SingletonLazy.getLazyInstance1();
        var lazySingleton2 = SingletonLazy.getLazyInstance1();

        System.out.println(lazySingleton1);
        System.out.println(lazySingleton2);

    }

}
